pkg/dataplane/baremetal.go

```diff
@@ -71,7 +71,9 @@ func DeployBaremetalSet(
 						}
 						ipPrefix, _ := ipNet.Mask.Size()
 						instanceSpec.CtlPlaneIP = fmt.Sprintf("%s/%d", res.Address, ipPrefix)
-						baremetalSet.Spec.CtlplaneGateway = *res.Gateway
+						if res.Gateway != nil {
```

Comment:
What is the use case for not having ctlplane gateway in the netconfig?  Do they use static routes for controlplane? If they have gateway on external network, it won't work because we download stuff (download-cache) before configuring networks on edpm nodes. If it's something that the user missed, then we should throw an error here...

Reply:
I'm not sure about it either. Could be useful for air-gaped environments maybe?

In any case we should log the fact that there is no gateway set. Even if it's not going to be an error.

Reply:
Way better, now we don't have to worry about it at all.

Reply:
> yeah, without the gateway, it fails:

That's because you've `defaultRoute: true` for ctlplane network in ospdns. I was wondering if there is an use-case with gateway only on external network that we need to address. That would not work unless we configure network on nodes before downloading container images/packages.